frame/state-trie-migration/src/lib.rs

```diff
@@ -0,0 +1,1514 @@
+				keys.len() as u32,
+				0,
+				MigrationCompute::Signed,
+			));
+			Ok(().into())
+		}
+
+		/// Migrate the list of child keys by iterating each of them one by one.
+		///
+		/// All of the given child keys must be present under one `top_key`.
+		///
+		/// This does not affect the global migration process tracker ([`MigrationProcess`]), and
+		/// should only be used in case any keys are leftover due to a bug.
+		#[pallet::weight(Pallet::<T>::dynamic_weight(child_keys.len() as u32, *total_size))]
+		pub fn migrate_custom_child(
+			origin: OriginFor<T>,
+			top_key: Vec<u8>,
+			child_keys: Vec<Vec<u8>>,
+			total_size: u32,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			// ensure they can pay more than the fee.
+			let deposit =
+				T::SignedDepositPerItem::get().saturating_mul((child_keys.len() as u32).into());
+			ensure!(T::Currency::can_slash(&who, deposit), "not enough funds");
+
+			let mut dyn_size = 0u32;
+			for child_key in &child_keys {
+				if let Some(data) =
+					sp_io::default_child_storage::get(Self::child_io_key(&top_key), &child_key)
+				{
+					dyn_size = dyn_size.saturating_add(data.len() as u32);
+					sp_io::default_child_storage::set(
+						Self::child_io_key(&top_key),
+						&child_key,
+						&data,
+					);
+				}
+			}
+
+			if dyn_size != total_size {
+				let (_imbalance, _remainder) = T::Currency::slash(&who, deposit);
+				debug_assert!(_remainder.is_zero());
+				return Err("Wrong witness data".into())
+			}
+
+			Self::deposit_event(Event::<T>::Migrated(
+				0,
+				child_keys.len() as u32,
+				MigrationCompute::Signed,
+			));
+			Ok(().into())
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn integrity_test() {
+			assert!(!T::UnsignedBackOff::get().is_zero(), "UnsignedBackOff should not be zero");
+		}
+
+		fn on_initialize(_: BlockNumberFor<T>) -> Weight {
+			if let Some(limits) = Self::auto_limits() {
+				let mut task = Self::migration_process();
+				task.migrate_until_exhaustion(limits);
+				let weight = Self::dynamic_weight(task.dyn_total_items(), task.dyn_size);
+
+				log!(
+					info,
+					"migrated {} top keys, {} child keys, and a total of {} bytes.",
+					task.dyn_top_items,
+					task.dyn_child_items,
+					task.dyn_size,
+				);
+				Self::deposit_event(Event::<T>::Migrated(
+					task.dyn_top_items as u32,
+					task.dyn_child_items as u32,
+					MigrationCompute::Auto,
+				));
+				MigrationProcess::<T>::put(task);
+
+				weight
+			} else {
+				T::DbWeight::get().reads(1)
+			}
+		}
+
+		fn offchain_worker(now: BlockNumberFor<T>) {
+			if Self::ensure_offchain_repeat_frequency(now).is_err() {
+				return
+			}
+
+			log!(debug, "started offchain worker thread.");
+			if let Some(chain_limits) = Self::unsigned_limits() {
+				let mut task = Self::migration_process();
+				if task.finished() {
+					log!(warn, "task is finished, remove `unsigned_limits`.");
```

Comment:
this will log a warning on every block until we remove the unsigned_limits.
Also I think we should remove it automatically once the migration is finished.

Reply:
I prefer keeping it in storage since it contains some metadata about how much data was migrated, which could potentially help us with debugging if anything goes wrong. If we delete it here the last snapshot of the task statet is lost. 

I do agree that the log was too much, made it debug.